src/org/zaproxy/zap/extension/alert/ExtensionAlert.java

```diff
@@ -433,9 +433,28 @@ public void displayAlert(Alert alert) {
     }
 
     public void updateAlertInTree(Alert originalAlert, Alert alert) {
-        if (Constant.isLowMemoryOptionSet()) {
+        if (getView() == null || Constant.isLowMemoryOptionSet()) {

```

Comment:
I actually meant the view check to be in the following if statement, we do want to update the tree model in the deamon too (just not in the EDT, since it's not needed).